### PR TITLE
[DEV-5138] Handling Non Normative Award Amount Scenarios (pt 3)

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
Passing lots of props to the same component to handle the award amounts viz.

**Technical details:**

Fair warning, the component I refactored still needs some serious luv. Rather than taking static objects, it really needs an array of objects. My apologies for this. Rather than extending the scope of this _even more_, I've just added a todo to that effect.

**New Props on RectanglePercentViz**
For over spent scenarios, we have an `improper` property. Based on the design, we have to do some un-intutive things to get the mark up in good shape for it.

The data inside `improper` is used to render the label, the bar's are adjusted to display according to the mock using the `barWidthOverrides` property.

**New Helper**
`determineFileCScenario` is invoked inside `determineSpendingScenario`, if it is ever not-normal, we return `insufficientData` such that the chart is not drawn. The following are considered non-normative scenarios:

- fileCObligation exceeds Current or Potential award amount
- fileCOutlay is greather than fileCObligation

If fileC data is zero, we treat that as normative as this will be the truly most normative case.

**JIRA Ticket:**
[DEV-5318](https://federal-spending-transparency.atlassian.net/browse/DEV-5318)

**SS:**

Exceeds Current:
image

Exceeds Potential:
image

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, helper functions, and models `if applicable`
`N/A` All `componentWillReceiveProps` and `componentWillMount` in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3277](https://federal-spending-transparency.atlassian.net/browse/DEV-3277)
`N/A`[API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A`[Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/_scss/pages/award/shared/awardAmountsSection/_charts.scss [hunk 1]
```diff
@@ -5,6 +5,8 @@ $new-potential: #AAC6E2;
 $file-c-outlay: #6E338E;
 $file-c-obligation: #B699C6;
 
+$old-obligated: #4773aa;
```
Comment (ebdabbs) [inline src/_scss/pages/award/shared/awardAmountsSection/_charts.scss:8]: `$color-cool-blue-light`
Would also eventually like to make the COVID purple colors into variables to ensure they are consistent across pages

File: src/_scss/pages/award/shared/awardAmountsSection/_charts.scss [hunk 2]
```diff
@@ -48,57 +57,62 @@ $new-potential: #AAC6E2;
...
-                background: repeating-linear-gradient(
-                    45deg,
-                    $color-cool-blue-light,
-                    $color-cool-blue-light rem(8),
-                    $color-gold rem(2),
-                    $color-gold rem(10)
-                );
+            &.idv-overspending,
+            &.contract-overspending {
+                border-top: .4rem solid #f1f1f1;
```
Comment (ebdabbs) [inline src/_scss/pages/award/shared/awardAmountsSection/_charts.scss:62]: `$color-gray-lightest`

File: src/_scss/pages/award/shared/awardAmountsSection/_charts.scss [hunk 3]
```diff
@@ -48,57 +57,62 @@ $new-potential: #AAC6E2;
...
+            .award-amounts-viz__bar-wrapper {
+                &.idv-extreme-overspending-potential-wrapper,
+                &.contract-extreme-overspending-potential-wrapper {
+                    border: 0.2rem solid #5b616b;
+                    background-color: #fff;
+                    .award-amounts-viz__bar .nested-obligations .award-amounts-viz__bar,
+                    .award-amounts-viz__bar .nested-obligations .award-amounts-viz__bar {
+                        &.contract-extreme-overspending-current,
+                        &.idv-extreme-overspending-current {
+                            border: solid 0.4rem #d6d7d9;
```
Comment (ebdabbs) [inline src/_scss/pages/award/shared/awardAmountsSection/_charts.scss:79]: `$color-gray-lighter`

File: src/js/components/award/shared/awardAmountsSection/charts/SharedBarComponents.jsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ const BarValue = ({
     number
 }) => (
     <div
-        style={style}
+        style={{ ...style, minWidth: '100px' }}
```
Comment (ebdabbs) [inline src/js/components/award/shared/awardAmountsSection/charts/SharedBarComponents.jsx:14]: Recommend adding a comment that this is to allow enough width for label text (we are not manipulating the line itself, which could be misleading about the data)

File: src/js/dataMapping/award/awardAmountsSection.js [hunk 5]
```diff
@@ -21,8 +21,16 @@ export const caresActSpendingCategories = [
 export const mockAwardIdsForCaresAct = [
     // contract
     'CONT_AWD_N0001917C0001_9700_-NONE-_-NONE-',
+    // exceeds current contract
```
Comment (ebdabbs) [inline src/js/dataMapping/award/awardAmountsSection.js:24]: Please create a Jira task to remove the mocked data before launch!

Comment (MaxwellKendall) [reply src/js/dataMapping/award/awardAmountsSection.js:25]: https://federal-spending-transparency.atlassian.net/browse/DEV-5434

Comment (ebdabbs) [reply src/js/dataMapping/award/awardAmountsSection.js:25]: Thank you!

**Pull request merged**